dtbase/backend/api/model/routes.py

```diff
@@ -25,15 +26,16 @@ def insert_model() -> Tuple[Response, int]:
         "name": <model_name:str>
     }
     """
-
     payload = request.get_json()
     error_response = check_keys(payload, ["name"], "/insert-model")
     if error_response:
         return error_response
-
-    models.insert_model(name=payload["name"], session=db.session)
+    try:
+        models.insert_model(name=payload["name"], session=db.session)
+    except IntegrityError:
+        return jsonify({"message": "Model exists already"}), 409
     db.session.commit()
-    return jsonify(payload), 201
+    return jsonify({"message": "Model inserted"}), 201
```

Comment:
Quite like to discuss this one @mhauru at our meeting today as it looks like it fundamentally changes the output of this function.